functions/logStripeEvent.js

```diff
@@ -0,0 +1,90 @@
+const functions = require('firebase-functions');
+const admin = require('firebase-admin');
+const logDonation = require('./logDonation');
+if (admin.apps.length === 0) {
+  admin.initializeApp();
+}
+
+const DEFAULTCPL = 1.0;
+
+exports.logPaymentIntent = functions.https.onRequest(async (req, res) => {
+  if (!req.body) {
+    res.status(501).send('no data supplied!');
+    return;
+  }
+  let event;
+  try {
+    event = req.body;
```

Comment:
Instead of putting the `req.body` assignment operator inside of a try/catch and performing a check above to validate the `body` property is present, couldn't we just rewrite lines 11-22 with 
```
  const event = req.body
  
  if (!req.body) {
    return res.status(400).send('no data supplied!');
  }
```
with the status code being 400 since the sender supplied the incorrect information?

Reply:
that's a much better way of doing it. originally there was a call to JSON.parse in there but it turns out firebase functions already does the parsing of the request body, so it was throwing an error on properly formatted data.